std/algorithm/iteration.d

```diff
@@ -4280,3 +4284,111 @@ unittest
          [1, 2, 0],
          [2, 1, 0]]));
 }
+
+/**
+Finds the mean (colloquially known as the average) of a range.
+This function has $(D O(r.length)) complexity.
+
+Params:
+    r = an input range
+
+Returns:
+    the mean of r
+*/
+auto mean(R)(R r) if (isInputRange!R && !isInfinite!R)
+in
+{
+    static if (hasLength!R)
+    {
+        assert(r.length > 0);
+    }
+}
+body
+{
+    import std.range.primitives : hasLength;
+
+    static if (hasLength!R)
+    {
+        return sum(r) / r.length;
+    }
+    else
+    {
+        import std.typecons : tuple;
+
+        immutable auto pair = reduce!((a, b) => tuple(a[0] + 1, a[1] + b))(tuple(0u, 0.0), r);
+        return pair[1] / pair[0];
+    }
+}
+
+///
+@safe unittest {
```

Comment:
Hate to be nitpicking on this again, but Phobos style guide requires opening braces to be on a new line.